src/components/ui/button.tsx

```diff
@@ -26,13 +27,17 @@ export const buttonVariants = cva(
 	},
 );
 
-type ButtonBaseComponent = HTMLElementTagNameMap['button'];
+type ButtonBaseComponent = HTMLElementTagNameMap["button"];
 
-type ButtonProps = React.ButtonHTMLAttributes<ButtonBaseComponent> & VariantProps<typeof buttonVariants>;
+type ButtonProps = React.ButtonHTMLAttributes<ButtonBaseComponent> &
+	VariantProps<typeof buttonVariants> & {
+		asChild?: boolean;
+	};
 
 export const Button = React.forwardRef<ButtonBaseComponent, ButtonProps>(
-	({className, variant, size, ...props}, ref) => {
-		return <button className={cn(buttonVariants({variant, size, className}))} ref={ref} {...props} />;
+	({ className, variant, size, asChild = false, ...props }, ref) => {
+		const Comp = asChild ? Slot : "button";
```

Comment:
In line with the comment left on the test file, let's keep the implementation and test in sync.